scaling/src/test/java/io/quarkus/ts/openshift/scaling/ScalingOpenShiftIT.java

```diff
@@ -0,0 +1,111 @@
+package io.quarkus.ts.openshift.scaling;
+
+import io.quarkus.ts.openshift.common.OpenShiftTest;
+import io.quarkus.ts.openshift.common.injection.TestResource;
+import io.quarkus.ts.openshift.common.util.OpenShiftUtil;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+
+import static io.restassured.RestAssured.given;
+import static javax.ws.rs.core.Response.Status.OK;
+import static javax.ws.rs.core.Response.Status.SERVICE_UNAVAILABLE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+@OpenShiftTest
+public class ScalingOpenShiftIT {
+
+    public static final String DC_NAME = "test-scaling";
+
+    @TestResource
+    private OpenShiftUtil openShiftUtil;
+
+    @AfterEach
+    public void scaleBack() {
+        openShiftUtil.scale(DC_NAME, 1);
+    }
+
+    @Test
+    public void scaleUpTest() {
+        given()
+                .when().get("/scaling")
+                .then()
+                .statusCode(OK.getStatusCode());
+
+        openShiftUtil.scale(DC_NAME, 2);
+
+        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(2);
+
+        Set<String> uniqueIds = Collections.newSetFromMap(new ConcurrentHashMap<>());
+        await().atMost(1, TimeUnit.MINUTES).untilAsserted(() -> {
+            String uniqueId =
+                    given()
+                            .when().get("/scaling")
+                            .then()
+                            .statusCode(OK.getStatusCode())
+                            .extract().asString();
+            uniqueIds.add(uniqueId);
+
+            assertThat(uniqueIds).hasSize(2);
+        });
+    }
+
+    @Test
+    public void scaleDownTest() {
+        openShiftUtil.scale(DC_NAME, 2);
+
+        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(2);
+
+        Set<String> uniqueIds = Collections.newSetFromMap(new ConcurrentHashMap<>());
+        await().atMost(1, TimeUnit.MINUTES).untilAsserted(() -> {
+            String uniqueId =
+                    given()
+                            .when().get("/scaling")
+                            .then()
+                            .statusCode(OK.getStatusCode())
+                            .extract().asString();
+            uniqueIds.add(uniqueId);
+
+            assertThat(uniqueIds).hasSize(2);
+        });
+
+        openShiftUtil.scale(DC_NAME, 1);
+
+        assertThat(openShiftUtil.countReadyReplicas(DC_NAME)).isEqualTo(1);
+
+        uniqueIds.clear();
+        for (int i = 0; i < 100; i++) {
```

Comment:
I would add few words why for loop is used instead of `await().atMost(1, TimeUnit.MINUTES).untilAsserted(()`

Reply:
I've added workflow to each test method.

Basically the thinking was to gain a slight confidence about the fact that the deployment was scaled down, and it wasn't an intermittent hiccup or such.